File: src/main/java/studio/karllang/karl/lib/FunctionManager.java
package studio.karllang.karl.lib;

import java.util.HashMap;

public final class FunctionManager {
    private static final HashMap<String, Function> functions = new HashMap<>();

    public static void addFunction(Function function) {
        functions.put(function.getName(), function);
    }

    public static Function getFunction(String name) {
        return functions.get(name);
    }

    public static boolean isFunction(String name) {
        return functions.containsKey(name);
    }

    public static void clear() {
        functions.clear();
    }

}

File: src/main/kotlin/studio/karllang/karl/lib/FunctionManager.kt
package studio.karllang.karl.lib

class FunctionManager {
    companion object {
        private val functions = HashMap<String, Function>()

        @JvmStatic
        fun addFunction(function: Function) {
            functions[function.getName()] = function
        }

        @JvmStatic
        fun getFunction(name: String): Function? {
            return functions[name]
        }

        @JvmStatic
        fun isFunction(name: String): Boolean {
            return functions.containsKey(name)
        }

        @JvmStatic
        fun clear() {
            functions.clear()
        }
    }
}
